tests/Operation/AggregateFunctionalTest.php

```diff
@@ -109,6 +109,17 @@ public function testTypeMapOption(array $typeMap = null, array $expectedDocument
         $this->assertEquals($expectedDocuments, $results);
     }
 
+    public function testExplainOption()
+    {
+        $this->createFixtures(3);
+
+        $pipeline = [['$match' => ['_id' => ['$ne' => 2]]]];
+        $operation = new Aggregate($this->getDatabaseName(), $this->getCollectionName(), $pipeline, ['explain' => true, 'typeMap' => ['root' => 'array', 'document' => 'array']]);
+        $results = iterator_to_array($operation->execute($this->getPrimaryServer()));
+
+        $this->assertArrayHasKey('queryPlanner', $results[0]['stages'][0]['$cursor']);
```

Comment:
I think we can simply assert that the count of `$results` is 1 and that a top-level `stages` field exists in the result document.

That should make this fairly resilient to the note in [Return Information on the Aggregation Operation](https://docs.mongodb.com/manual/reference/command/aggregate/#return-information-on-the-aggregation-operation):

> The explain output is subject to change between releases.

Reply:
Also, I think we can remove the `typeMap` option above. The `ArrayHasKey` constraint [does support the ArrayAccess interface](https://github.com/sebastianbergmann/phpunit/blob/7.0.1/src/Framework/Constraint/ArrayHasKey.php#L65), which our default BSONDocument class implements. Looking at the git history of the constraint class, that functionality goes back to PHPUnit 3.x, so we can rely on it for all variants in the test matrix.

Reply:
If I remove the `typeMap` option, the result returned is an array wrapped in a `stdClass` object so I still need `['typeMap' => ['root' => 'array']]`.

Reply:
Aye, I forgot we're not inheriting the Collection's type map here.